scripts/tests_coverage.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+# Butterfly build root
+BUTTERFLY_BUILD_ROOT=$1
+
+# Test Butterfly build root
+if [ ! -d $BUTTERFLY_BUILD_ROOT/api ]; then
+    echo "Please set butterfly's build root as parameter"
+    exit 1
+fi
+
+client_obj_dir=$BUTTERFLY_BUILD_ROOT/api/client/CMakeFiles/api_client.dir/
+server_obj_dir=$BUTTERFLY_BUILD_ROOT/api/server/CMakeFiles/api_server.dir/
+
+# Test coverage data files
+ret=$(find $BUTTERFLY_BUILD_ROOT/api/ -name *.gcda | wc -l)
+if [ $ret -eq 0 ]; then
+    echo -e "
+Coverage files (*.gcda) not found
+in $client_obj_dir 
+or 
+in $server_obj_dir
+Make sure you did that in your build directory:
+- cmake .. -DDEBUG=ON
+- make 
+run some tests before running coverage test.
+"
+    exit 1
+fi
+
+function check_error {
+output=$1
+if [ $output -ne 0 ]; then
+    echo COVERAGE TEST FAILED!!
+    exit 1
+fi
+}
+
+function clean_coverage_data {
+    find $BUTTERFLY_BUILD_ROOT/api/ \( -name *.gcda \) -exec rm {} \;
+    check_error $?
+    rm client.info server.info
+    check_error $?
+}
+
+lcov -c -d $client_obj_dir -o client.info
```

Comment:
This binary should be checked before
something like `lcov -h || echo "Error: lcov command not found" && exit 1`